hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/api/graph/EdgeAPI.java

```diff
@@ -291,6 +291,19 @@ public String list(@Context GraphManager manager,
         HugeGraph g = graph(manager, graph);
 
         GraphTraversal<?, Edge> traversal;
+
+        // Optimize edge query. Return early when the EdgeLabel does not exist.
+        if (label != null && !g.existsEdgeLabel(label)) {
```

Comment:
could you please move these logic into [GraphTransaction.optimizeQuery()](https://github.com/apache/incubator-hugegraph/blob/57cd0e8dd7035e7dd6cb4581c1ed00ee1b2c83f4/hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/tx/GraphTransaction.java#L1450), so that it can also be used by Gremlin query.

Reply:
我在测试的时候发现，当 EdgeLabel 不存在时，会在上层代码的 makeQuery() 方法检测出来（E.checkArgument），不会进入 GraphTransaction.optimizeQuery 方法，在 GraphTransaction.optimizeQuery 中添加这个检测，是否重复了？

具体的堆栈如下：
  edgeLabel:891, StandardHugeGraph (org.apache.hugegraph)
  getLabelId:167, SchemaLabel (org.apache.hugegraph.schema)
  convSysValueIfNeeded:711, TraversalUtil (org.apache.hugegraph.traversal.optimize)
  convCompare2SyspropRelation:442, TraversalUtil (org.apache.hugegraph.traversal.optimize)
  convCompare2Relation:430, TraversalUtil (org.apache.hugegraph.traversal.optimize)
  convHas2Condition:358, TraversalUtil (org.apache.hugegraph.traversal.optimize)
  fillConditionQuery:320, TraversalUtil (org.apache.hugegraph.traversal.optimize)
  **makeQuery**:155, HugeGraphStep (org.apache.hugegraph.traversal.optimize)
  edges:138, HugeGraphStep (org.apache.hugegraph.traversal.optimize)
  lambda$new$0:66, HugeGraphStep (org.apache.hugegraph.traversal.optimize)

Reply:
We aim to optimize the scenario of adjacent edges query: such as `g.V().out('follow')`, if the `g.V()` return a person v1 and a movie v2, the movie vertex v2 has no 'follow' edge, so we don't need to query it from the storage.

Reply:
Okay, I'll add this part of the code later.